incubator/hnc/internal/reconcilers/anchor.go

```diff
@@ -106,6 +114,25 @@ func (r *AnchorReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	return ctrl.Result{}, r.writeInstance(ctx, log, inst)
 }
 
+// upgradeSubnamespaceOfAPI replaces the deprecated "subnamespaceOf" annotation
+// with the new "subnamespace-of" annotation.
+func (r *AnchorReconciler) upgradeSubnamespaceOfAPI(ctx context.Context, log logr.Logger, inst *corev1.Namespace, sOf string) error {
+	// Remove old annotation.
+	a := inst.GetAnnotations()
+	delete(a, api.DeprecatedSubnamespaceOf)
+	inst.SetAnnotations(a)
+
+	// Add new annotation.
+	metadata.SetAnnotation(inst, api.SubnamespaceOf, sOf)
```

Comment:
Here, it looks like if both are present, the old one will overwrite the new one. Is that intended?

Reply:
Ah, it should be that if both exists, we will just remove the deprecated annotation directly, right?

Reply:
That would make sense, I think

Reply:
Done.